src/NServiceBus.TransactionalSession/TransactionalSessionBase.cs

```diff
@@ -45,6 +44,11 @@ public ISynchronizedStorageSession SynchronizedStorageSession
 
         public virtual Task Open(OpenSessionOptions options = null, CancellationToken cancellationToken = default)
         {
+            if (IsOpen)
```

Comment:
alternatively, we could just return without throwing an exception

Reply:
I think throwing is safer.